src/content/en/fundamentals/native-hardware/capturing-images/index.md

```diff
@@ -2,116 +2,242 @@ project_path: /web/_project.yaml
+If you are already adding in the ability to upload a file, there are a couple of easy ways that you
+can make the user experience a little richer.
+
+The first is to add a drop target to your page that allows the user to drag in a file from the
+desktop or another application.
+
+    <div id="target">You can drag an image file here</div>
     <script>
-      var camera = document.getElementById('camera');
-      var frame = document.getElementById('frame');
+      const target = document.getElementById('target');
 
-      camera.addEventListener('change', function(e) {
-        var file = e.target.files[0]; 
-        // Do something with the image file.
-        frame.src = URL.createObjectURL(file);
+      target.addEventListener('drop', (e) => {
+        e.stopPropagation();
+        e.preventDefault();
+
+        doSomethingWithFiles(e.dataTransfer.files);
+      });
+
+      target.addEventListener('dragover', (e) => {
+        e.stopPropagation();
+        e.preventDefault();
+
+        e.dataTransfer.dropEffect = 'copy';
       });
     </script>
 
-Once you have access to the file you can do anything you want with it. For 
-example, you can:
+Similar to the file input, you can get a `FileList` object from the `dataTransfer.files` property of
+the `drop` event;
+
+The `dragover` event handler let's us signal to the user what will happen if they drop the file they
+are dragging onto the target.
+
+Drag and drop has been around for a long time and is well supported by the major browsers.
+
+### Paste from clipboard
+The final way to get an existing image file is from the clipboard. The code for this is very simple,
+but the user experience is a little harder to get right.
+
+    <textarea id="target">Paste an image here</textarea>
+    <script>
+      const target = document.getElementById('target');
+
+      target.addEventListener('paste', (e) => {
+        e.preventDefault();
+        doSomethingWithFiles(e.clipboardData.files);
+      });
+    </script>
+
+(`e.clipboardData.files` is yet another FileList object.)
+
+The tricky part with the clipboard API is that, for full cross-browser support, the target element
+needs to be both selectable and editable. Both `<textarea>` and `<input type="text">` fit the bill
+here, but are both also obviously designed for editing text.
+
+It can be difficult to make this work smoothly if you don't want to user to also be able to input
+text. Tricks like having a hidden input that gets selected when you click on some other element can
+make maintaining accessibility harder.
+
+### Handling a FileList object
+Since most of the above methods produce a `FileList`, we should talk a little about what that is.
+
+A `FileList` is similar to an `Array`. It has numeric keys and a `length` property, but it isn't
+*actually* an array. There are no array methods, like `forEach` or `pop`, and it isn't iterable.
+Even `Array.from(fileList)` doesn't work correctly. If you want to iterate over all of the files you
```

Comment:
What is not working with `Array.from(fileList)`? It looks like it works fine for me.